Index: sources/lib/locking/cluster_locking.c
===================================================================
--- sources.orig/lib/locking/cluster_locking.c	2011-06-30 22:02:44.176694657 +0200
+++ sources/lib/locking/cluster_locking.c	2011-06-30 22:02:46.364705507 +0200
@@ -25,6 +25,7 @@
 #include "locking.h"
 #include "locking_types.h"
 
+#include <linux/limits.h>
 #include <stddef.h>
 #include <sys/socket.h>
 #include <sys/un.h>
