diff -Nur src/ethtool.c src.new/ethtool.c
--- src/ethtool.c	2008-12-02 21:03:00.000000000 +0100
+++ src.new/ethtool.c	2008-12-20 12:27:57.000000000 +0100
@@ -1757,8 +1757,7 @@
 	wol->supported = 0;
 	wol->wolopts = 0;
 
-	if (!(adapter->flags & FLAG_HAS_WOL) ||
-	    !device_can_wakeup(&adapter->pdev->dev))
+	if (!(adapter->flags & FLAG_HAS_WOL))
 		return;
 
 	wol->supported = WAKE_UCAST | WAKE_MCAST |
@@ -1793,10 +1792,7 @@
 {
 	struct e1000_adapter *adapter = netdev_priv(netdev);
 
-	if (!(adapter->flags & FLAG_HAS_WOL) ||
-	    !device_can_wakeup(&adapter->pdev->dev) ||
-	    (wol->wolopts & ~(WAKE_UCAST | WAKE_MCAST | WAKE_BCAST |
-	                      WAKE_MAGIC | WAKE_PHY | WAKE_ARP)))
+	if (!(adapter->flags & FLAG_HAS_WOL))
 		return -EOPNOTSUPP;
 
 	/* these settings will always override what we currently have */
@@ -1815,7 +1811,6 @@
 	if (wol->wolopts & WAKE_ARP)
 		adapter->wol |= E1000_WUFC_ARP;
 
-	device_set_wakeup_enable(&adapter->pdev->dev, adapter->wol);
 
 	return 0;
 }
diff -Nur src/netdev.c src.new/netdev.c
--- src/netdev.c	2008-12-02 21:03:00.000000000 +0100
+++ src.new/netdev.c	2008-12-20 12:27:45.000000000 +0100
@@ -5519,7 +5519,6 @@
 
 	/* initialize the wol settings based on the eeprom settings */
 	adapter->wol = adapter->eeprom_wol;
-	device_set_wakeup_enable(&adapter->pdev->dev, adapter->wol);
 
 	/* save off EEPROM version number */
 	e1000_read_nvm(&adapter->hw, 5, 1, &adapter->eeprom_vers);
